File: port/src/main/Xml/Core/ClassScannerTest.cs
#region Using directives
using SimpleFramework.Xml;
using System;
#endregion
namespace SimpleFramework.Xml.Core {
   public class ClassScannerTest : TestCase {
      [Root]
      [Order(Elements={"a", "b"}, Attributes={"A", "B"})]
      [Namespace(Prefix="prefix", Reference="http://domain/reference")]
      private static class Example {
         [Commit]
         public void Commit() {
            return;
         }
         [Validate]
         public void Validate() {
            return;
         }
      }
      public void TestClassScanner() {
         ClassScanner scanner = new ClassScanner(Example.class);
         assertNotNull(scanner.getRoot());
         assertNotNull(scanner.getOrder());
      }
   }
}

File: src/test/java/org/simpleframework/xml/core/ClassScannerTest.java
package org.simpleframework.xml.core;

import junit.framework.TestCase;

import org.simpleframework.xml.Namespace;
import org.simpleframework.xml.Order;
import org.simpleframework.xml.Root;
import org.simpleframework.xml.stream.Format;

public class ClassScannerTest extends TestCase {
   
   @Root
   @Order(elements={"a", "b"}, attributes={"A", "B"})
   @Namespace(prefix="prefix", reference="http://domain/reference")
   private static class Example {
      
      @Commit
      public void commit() {
         return;
      }
      
      @Validate
      public void validate() {
         return;
      }
   }

   public void testClassScanner() throws Exception {
      ClassScanner scanner = new ClassScanner(Example.class, new Format());
      
      assertNotNull(scanner.getRoot());
      assertNotNull(scanner.getOrder());
   }
}
